rules/declaration-block-properties-flexible-order/index.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+const stylelint = require('stylelint');
+const _ = require('lodash');
+const utils = require('../../utils');
+const declarationBlockPropertiesSpecifiedOrder = require('../declaration-block-properties-specified-order');
+
+const ruleName = utils.namespace('declaration-block-properties-flexible-order');
+
+const messages = stylelint.utils.ruleMessages(ruleName, {
+	expected: (first, second) => `Expected ${first} to come before ${second}`,
+});
+
+function createExpectedOrder(input) {
+	const order = {};
+	let expectedPosition = 0;
+
+	input.forEach((item) => {
+		expectedPosition += 1;
+
+		order[item] = {
+			expectedPosition,
+		};
+	});
+
+	appendGroup(input);
+
+	function appendGroup(items) {
+		items.forEach((item) => appendItem(item, false));
+	}
+
+	function appendItem(item, inFlexibleGroup) {
+		if (_.isString(item)) {
+			// In flexible groups, the expectedPosition does not ascend
+			// to make that flexibility work;
+			// otherwise, it will always ascend
+			if (!inFlexibleGroup) {
+				expectedPosition += 1;
+			}
+
+			order[item] = { expectedPosition };
+
+			return;
+		}
+
+		if (!item.order || item.order === 'strict') {
+			appendGroup(item.properties);
+		} else if (item.order === 'flexible') {
+			expectedPosition += 1;
+			item.properties.forEach((property) => {
+				appendItem(property, true);
+			});
+		}
+	}
+
+	return order;
+}
+
+function validatePrimaryOption(actualOptions) {
```

Comment:
Shouldn't check for `order` property. But [should check](https://github.com/stylelint/stylelint/blob/16e8cc79f8d08cd9502fd9ab58f9d137c08709b5/lib/rules/declaration-block-properties-order/index.js#L272-L281) that every item is `string` or object with `properties` property.